// scaffold/ngController/Content.js
"use strict";

exports.getText = function () {
    return '\
"use strict";\n\n\
angular.module("meanStack.{{domainNameInSmallLetter}}.controllers", [])\n\
    .controller("{{domainName}}ListController", ["$scope", "$http", "$routeParams", function ($scope, $http, $routeParams) {\n\
        var offset = parseInt($routeParams.offset, 10);\n\
        var max = parseInt($routeParams.max, 10);\n\
        $http.get("/{{domainNameInSmallLetter}}/list/" + $routeParams.max + "/" + $routeParams.offset)\n\
            .success(function (data, status) {\n\
                if (status == 200) {\n\
                    $scope.{{domainNameInSmallLetter}}s = data.{{domainNameInSmallLetter}}s;\n\
                    $scope.total = parseInt(data.total, 10);\n\
                    $scope.listParams = {\n\
                        max: max,\n\
                        nextOffset: ((offset + max) >= $scope.total) ? -1 : offset + max,\n\
                        previousOffset: offset == 0 ? -1 : offset - max\n\
                    };\n\
                } else {\n\
                    alert(data);\n\
                }\n\
            })\n\
            .error(function (data, status) {\n\
                alert(data, status);\n\
            });\n\
        $scope.remove{{domainName}} = function (index, id) {\n\
            if (confirm("Are you sure?")) {\n\
                $http.delete("/{{domainNameInSmallLetter}}/delete/" + id)\n\
                    .success(function (data, status) {\n\
                        if (status == 200) {\n\
                            $scope.{{domainNameInSmallLetter}}s.splice(index, 1);\n\
                            $scope.$apply();\n\
                            alert("{{domainName}} Deleted");\n\
                        } else {\n\
                            alert(data);\n\
                        }\n\
                    })\n\
                    .error(function (data, error) {\n\
                        alert(data);\n\
                    });\n\
            }\n\
        };\n\
    }])\n\
    .controller("{{domainName}}ShowController", ["$scope", "$http", "$routeParams", "$location", function ($scope, $http, $routeParams, $location) {\n\
        $http.get("/{{domainNameInSmallLetter}}/get/" + $routeParams.id)\n\
            .success(function (data, status) {\n\
                if (status == 200) {\n\
                    $scope.{{domainNameInSmallLetter}} = data;\n\
                } else {\n\
                    alert(data);\n\
                }\n\
            })\n\
            .error(function (data, status) {\n\
                alert(data, status)\n\
            });\n\
        $scope.remove{{domainName}} = function (id) {\n\
            if (confirm("Are you sure?")) {\n\
                $http.delete("/{{domainNameInSmallLetter}}/delete/" + id)\n\
                    .success(function (data, status) {\n\
                        if (status == 200) {\n\
                            $location.path("/{{domainNameInSmallLetter}}/list/10/0");\n\
                            $scope.$apply();\n\
                        } else {\n\
                            alert(data);\n\
                        }\n\
                    })\n\
                    .error(function (data, error) {\n\
                        alert(data);\n\
                    });\n\
            }\n\
        };\n\
    }])\n\
    .controller("{{domainName}}CreateController", ["$scope", "$http", "$location", function ($scope, $http, $location) {\n\
        $scope.save{{domainName}} = function () {\n\
            $http.post("/{{domainNameInSmallLetter}}/save", serialize({\n\
                username: $scope.{{domainNameInSmallLetter}}.username,\n\
                password: $scope.{{domainNameInSmallLetter}}.password\n\
            }), {\n\
                headers: {\n\
                    "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8"\n\
                }\n\
            })\n\
                .success(function (data, status) {\n\
                    if (status == 201) {\n\
                        $location.path("/{{domainNameInSmallLetter}}/list/10/0");\n\
                    } else {\n\
                        alert(data);\n\
                    }\n\
                })\n\
                .error(function (data, status) {\n\
                    alert(data, status)\n\
                });\n\
        };\n\
    }])\n\
    .controller("{{domainName}}EditController", ["$scope", "$http", "$location", "$routeParams", function ($scope, $http, $location, $routeParams) {\n\
            $http.get("/{{domainNameInSmallLetter}}/get/" + $routeParams.id)\n\
                .success(function (data, status) {\n\
                    if (status == 200) {\n\
                        $scope.{{domainNameInSmallLetter}} = data;\n\
                    } else {\n\
                        alert(data);\n\
                    }\n\
                })\n\
                .error(function (data, status) {\n\
                    alert(data, status)\n\
                });\n\
            $scope.update{{domainName}} = function () {\n\
                $http.put("/{{domainNameInSmallLetter}}/update", serialize({\n\
                    _id: $scope.{{domainNameInSmallLetter}}._id,\n\
                    username: $scope.{{domainNameInSmallLetter}}.username,\n\
                    password: $scope.{{domainNameInSmallLetter}}.password\n\
                }), {\n\
                    headers: {\n\
                        "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8"\n\
                    }\n\
                }).success(function (data, status) {\n\
                        if (status == 202) {\n\
                            $location.path("/{{domainNameInSmallLetter}}/show/" + $routeParams.id);\n\
                        } else {\n\
                            alert(data);\n\
                        }\n\
                    })\n\
                    .error(function (data, status) {\n\
                        alert(data, status)\n\
                    });\n\
            };\n\
            $scope.remove{{domainName}} = function (id) {\n\
                if (confirm("Are you sure?")) {\n\
                    $http.delete("/{{domainNameInSmallLetter}}/delete/" + id)\n\
                        .success(function (data, status) {\n\
                            if (status == 200) {\n\
                                $location.path("/{{domainNameInSmallLetter}}/list/10/0");\n\
                                $scope.$apply();\n\
                            } else {\n\
                                alert(data);\n\
                            }\n\
                        })\n\
                        .error(function (data, error) {\n\
                            alert(data);\n\
                        });\n\
                }\n\
            };\n\
        }]);'
};
